### DISCUSSION TITLE
Can Coqui TTS be deployed on iOS and Android?

### DISCUSSION THREAD

**Discussion by ethanfa** [coqui-ai/TTS#2306] [General Q&A] [answered]
Hi,

Coqui TTS is awesome! It seems that it can also be used commercially.
I was thinking of using it as TTS on iOS or Android, and I wondered:
1. Can Coqui TTS be deployed on iOS and Android, especially iPhone and Samsung phones? Does it work for them?
2. If I only use it for TTS, without any modification to the code, do I need to publish the rest of my code that uses Coqui?
3. Does it come with a pre-trained model I can use for iPhone and Samsung phones?

Thanks!

**Comment by erogol** [comment, marked as answer]
Yes in theory but we don't have official support.

**Comment by csukuangfj** [comment]
I want to share that [sherpa-onnx](http://github.com/k2-fsa/sherpa-onnx) supports running
VITS models from Coqui on Android.

The support for iOS will be ready in the coming week.

Please see
https://github.com/coqui-ai/TTS/discussions/3194
for details.

**Reply by mablue** [reply]
Thanks it will be very useful for Persians blind community.

**Reply by csukuangfj** [reply]
@mablue 

FYI: We have converted the Persian VITS TTS model from [Mimic3](https://github.com/MycroftAI/mimic3-voices) to [sherpa-onnx](https://github.com/k2-fsa/sherpa-onnx) and have built Android TTS APKs for it.

By the way, the underlying implementation is in C++ and it uses [piper-phonemize](https://github.com/rhasspy/piper-phonemize) to convert words to phonemes.


Please see
- https://k2-fsa.github.io/sherpa/onnx/tts/apk-engine.html
- https://k2-fsa.github.io/sherpa/onnx/tts/apk.html

For the TTS engine APK, you can use it to replace the system TTS engine.

|standalone tts|tts engine|
|---|---|
|![Screenshot 2024-01-14 at 19 30 07](https://github.com/k2-fsa/sherpa-onnx/assets/5284924/1067570f-7f62-475b-8611-8bdcc9797dff)|![Screenshot 2024-01-14 at 19 31 00](https://github.com/k2-fsa/sherpa-onnx/assets/5284924/f20c3b77-5cc9-4d25-9fc5-04b66625280f)|

**Comment by mablue** [comment]
Hello. I tested it, the sound quality is very good. sHe's just a little slow, and if he works on his accent and pronunciation, she won't have anything left.


its beautiful
 but code of persian apk have google play protect And didnt install

it will have a universal version for all phones its hard to find out architecture than install the spacific version for blinds.

i tested persian but when there is an english word inside persian words it cant read it good!

**Reply by csukuangfj** [reply]
> Hello. I tested it, the sound quality is very good. sHe's just a little slow, and if he works on his accent and pronunciation, she won't have anything left.
> 
> 
> its beautiful
>  but code of persian apk have google play protect And didnt install
> 
> it will have a universal version for all phones its hard to find out architecture than install the spacific version for blinds.
> 
> i tested persian but when there is an english word inside persian words it cant read it good!

Have you ever tried a Persian TTS App that could read the English word?  It's expected for the current APK.

**Reply by mablue** [reply]
yes I expected. we use google sentetic tts in urdu language and or some other tts like dariush voice in vocalizer tts app and them can read english words two. also still couse speed everyone using espeak in iran cuz its very fast about(400wpm)and can read multilangual persian and eng  but the problem is robotic voice...
The current Persian language used in Iran is a mixture of English and Persian...
its an example: download dariush voice inside app and see how it can read persian and english easily:
https://t.me/JalebLog/5?single
biggest problem: its closed sorce and  moded and dictionary is not updateable for us.
if shapra have an small model to start fast it will be good:
```
امیر حسین:
این متن به گفتار صدای فوق العاده ای داره ولی دیکشنری و خواندن لغاتش جالب نیست و قابلیت اجرا در پس‌زمینه اش هم ضعیفه امیدوارم در آبدیتهای بعدی این مشکلات مرتفع بشه

Masoud:
فیدبک ها رو به تیم سازنده انتقال خواهم داد امیدوارم بتونن حلش کنن تشکر از نظرات خوبتون

امیر حسین:
خواهش میکنم ضمنا اگر تاخیر خوانشش هم کمتر بشه عالی میشه

Masoud:
حتما بررسی میشه سرعتو افزایش دادین با next-gen kaldi tts ؟؟

امیر حسین:
اون سرعت خانش را تنظیم می‌کند و فکر نکنم بتونه مشکل تاخیر خوانش را برطرف کنه

Masoud:
تاخیر به خاطر پردازش هست بله...

چون فکر میکنه به کلمات بعد میخونه مثل آدم و ابن فکر کردن زمان میبره 😂
```

people loves shapra with haniye voice but if it can start fast to reading with screenreaders like google talkback or commentary screen readers

**Reply by mablue** [reply]
some other feedbacks 
https://github.com/karim23657/Persian-tts-coqui/issues/33#issuecomment-1897877277

**Comment by malisharif67-collab** [comment]
python.org
